docs/en/creating-widgets/supplemental.md

```diff
@@ -343,6 +343,60 @@ Dojo uses a virtual node's key to uniquely identify a specific instance when re-
 
 > **Note:** Virtual node `keys` should be consistent across multiple render function invocations. Generating different keys for what should be the same output node within every render call [is considered an anti-pattern in Dojo application development](/learn/creating-widgets/best-practice-development#the-virtual-dom), and should be avoided.
 
+## Defining widget `key`s
+
+Traditionally the widget's `key` property is used by the Dojo rendering engine to uniquely identify and track widgets across renders. However updating the `key` property is also an effective way to guarantee that during the next render Dojo's rendering engine will recreate the widget instead of reusing the previous instance. In doing so this means all previous state will be reset widget and can be useful when working with widgets that manage logic based on a widget property.
```

Comment:
```suggestion
Traditionally the widget's `key` property is used by the Dojo rendering engine to uniquely identify and track widgets across renders. However, updating the `key` property is also an effective way to guarantee that during the next render Dojo's rendering engine will recreate the widget instead of reusing the previous instance. When recreating the widget all previous state will get reset. This behavior is useful when working with widgets that manage logic based on the value of a widget property.
```